pkg/service/controller.go

```diff
@@ -137,6 +137,19 @@ func (c *ControllerService) ControllerPublishVolume(
 		return nil, err
 	}
 
+	_, err = diskAttachmentByVmAndDisk(conn, req.NodeId, req.VolumeId)
+	if err != nil {
+		// If attachment was not found we are good to
+		_, ok := err.(*AttachmentNotFoundError)
```

Comment:
I think it is better to use `errors.Is()` or `errors.As()` here because it checks wrapped errors too. Not an issue now, but later with some refactoring this could become relevant.

Reply:
Thanks, I tried it previously,  but I think it doesn't apply here since the error isn't wrapped (and there is no need to wrap it as this error is actually desired in this case)

Reply:
It's more of a nitpick, really. I'm just thinking about the future when somebody is working on this code. If the error will be wrapped in the future using the `errors` methods make the code more resilient.

Reply:
I'm ok with nits :) do you have an example of how it would look in this case? Because as I understand to use `errors` I'd need to create an instance of the error struct, but perhaps I'm missing something

Reply:
From memory it should work like this:

```go
notFoundError := &AttachmentNotFoundError{}
if errors.As(err, &notFoundError) {
    // ...
}
```

Reply:
yes, like I said, it requires an additional instance which looks weird, but I changed it anyway

Reply:
Alternatively, you can also store the error in a variable and use `errors.Is`, that is less quirky.